src/amp-story-player/page-scroller.js

```diff
@@ -0,0 +1,229 @@
+   * @param {number} timeStamp
+   */
+  onTouchEnd(timeStamp) {
+    this.touchEventState_.touchEndTimeMs = timeStamp;
+
+    if (!this.scrollState_.meetsDeltaYThreshold) {
+      return;
+    }
+
+    const timeFromLastTouchMove =
+      this.touchEventState_.touchEndTimeMs -
+      this.touchEventState_.touchMoveTimeMs;
+
+    this.scrollState_.offsetPx = this.calculateOffset_();
+
+    if (
+      timeFromLastTouchMove < this.scrollState_.moveTimeThresholdMs &&
+      Math.abs(this.scrollState_.offsetPx) > this.scrollState_.offsetThresholdPx
+    ) {
+      // If timefromLastTouchMove is low enough and the offset is above the
+      // threshold, (re)set the scroll parameters to include momentum.
+      this.scrollState_.durationMs = this.win_./*OK*/ innerHeight * 1.2;
+      this.scrollState_.isRunning = true;
+
+      requestAnimationFrame((timestamp) => {
+        this.scrollState_.startTimeMs = timestamp;
+        this.scrollState_.startY = this.win_./*OK*/ scrollY;
+        this.scrollOnNextTick_(timestamp);
+      });
+    }
+
+    this.scrollState_.multiplier = 1;
+    this.touchEventState_.startY = 0;
+    this.scrollState_.deltaY = 0;
+  }
+
+  /**
+   * Calculates pixel offset and direction to be scrolled by getting current
+   * acceleration and preset limits.
+   * @private
+   * @return {number}
+   */
+  calculateOffset_() {
+    const maxOffset =
+      this.win_./*OK*/ innerHeight * this.scrollState_.speedLimit;
+
+    let offset =
+      Math.pow(this.scrollState_.acceleration, 2) *
+      this.win_./*OK*/ innerHeight;
+
+    offset = Math.min(maxOffset, offset);
+
+    offset *=
+      this.scrollState_.deltaY > 0
+        ? -this.scrollState_.multiplier
+        : this.scrollState_.multiplier;
+
+    return offset;
+  }
+
+  /**
+   * Calculates scrolling position at each frame. Stops once durationMs is met
+   * or scrollState stops running.
+   * @param {number} timeStamp
+   * @private
+   */
+  scrollOnNextTick_(timeStamp) {
+    const runTime = timeStamp - this.scrollState_.startTimeMs;
+
+    if (runTime > this.scrollState_.durationMs) {
+      return;
+    }
+
+    const progress = this.easeOutQuartFn_(
+      runTime / this.scrollState_.durationMs
+    );
+
+    const scrollDelta = progress * this.scrollState_.offsetPx;
+
+    const scrollForThisTick = this.scrollState_.startY + scrollDelta;
+
+    if (!this.scrollState_.isRunning) {
+      cancelAnimationFrame(
+        requestAnimationFrame(this.scrollOnNextTick_.bind(this))
+      );
+    } else {
+      this.win_.scroll(0, scrollForThisTick);
+      requestAnimationFrame(this.scrollOnNextTick_.bind(this));
+    }
+  }
+
+  /**
+   * Ease out quart function.
+   * @param {number} pTimeElapsed Percentage of time elapsed.
+   * @return {number} Percentage progress, value between 0 and 1.
+   * @private
+   */
+  easeOutQuartFn_(pTimeElapsed) {
+    return 1 - --pTimeElapsed * pTimeElapsed * pTimeElapsed * pTimeElapsed;
```

Comment:
What's the -- before the pTimeElapsed?

Reply:
Prefix decrement operator 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Decrement

Reply:
Ohh I forgot that existed in JS hehe